assets/helpers/payPalExpressCheckout/payPalExpressCheckout.js

```diff
@@ -113,15 +116,43 @@ function setup(dispatch: Function, getState: () => CombinedState, callback: Func
         commit: true,
 
         // This function is called when user clicks the PayPal button.
-        payment: setupPayment(dispatch, getState()),
+
+        payment: () => {
+
+          const htmlElement = document.getElementById(payPalId);
+
+          const dataset = htmlElement ? htmlElement.dataset : {};
+
+          const currentAmount = Number(dataset.amount) || amount;
+          const currentCurrency = ((dataset.currency: any): IsoCurrency) || currency;
+          const currentBillingPeriod = dataset.billingPeriod || billingPeriod;
+          const currentCsrfToken = dataset.csrfToken || csrfToken;
+
+          return setupPayment(
+            currentAmount,
+            currentBillingPeriod,
+            currentCurrency,
+            currentCsrfToken,
+          ).catch((err) => {
+            failureCallback(err);
+            throw err;
+          });
+        },
 
         // This function is called when the user finishes with PayPal interface (approves payment).
         onAuthorize,
       };
-      const payPalId = 'component-paypal-button-checkout';
+
       const htmlElement = document.getElementById(payPalId);
       const elementCount: ?number = htmlElement ? htmlElement.childElementCount : null;
 
+      if (htmlElement) {
+        htmlElement.dataset.amount = String(amount);
```

Comment:
My impression is this line will change the attribute `data-amount` of that element. Do you want to do that or change the value of the element?

Reply:
What do you mean by "change the value of the element"?

I intend to change the `data-amount`, so that it can be read on line 126.